File: ble-livedata/src/main/java/no/nordicsemi/android/ble/livedata/BondingStateLiveData.java
package no.nordicsemi.android.ble.livedata;

import android.bluetooth.BluetoothDevice;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import no.nordicsemi.android.ble.livedata.state.BondState;
import no.nordicsemi.android.ble.observer.BondingObserver;

class BondingStateLiveData extends LiveData<BondState> implements BondingObserver {

	BondingStateLiveData() {
		setValue(BondState.NotBonded.INSTANCE);
	}

	@Override
	public void onBondingRequired(@NonNull final BluetoothDevice device) {
		setValue(BondState.Bonding.INSTANCE);
	}

	@Override
	public void onBonded(@NonNull final BluetoothDevice device) {
		setValue(BondState.Bonded.INSTANCE);
	}

	@Override
	public void onBondingFailed(@NonNull final BluetoothDevice device) {
		setValue(BondState.NotBonded.INSTANCE);
	}
}

File: ble-livedata/src/main/java/no/nordicsemi/android/ble/livedata/BondingStateLiveData.kt
package no.nordicsemi.android.ble.livedata

import android.bluetooth.BluetoothDevice
import androidx.lifecycle.LiveData
import no.nordicsemi.android.ble.livedata.state.BondState
import no.nordicsemi.android.ble.observer.BondingObserver

@Suppress("unused")
internal class BondingStateLiveData: LiveData<BondState>(BondState.NotBonded), BondingObserver {

    override fun onBonded(device: BluetoothDevice) {
        value = BondState.Bonded
    }

    override fun onBondingFailed(device: BluetoothDevice) {
        value = BondState.NotBonded
    }

    override fun onBondingRequired(device: BluetoothDevice) {
        value = BondState.Bonding
    }

}